src/main/java/org/springframework/data/convert/PropertyValueConverterRegistrar.java

```diff
@@ -0,0 +1,191 @@
+	 *
+	 * @param type the domain type to obtain the property from
+	 * @param path the property name.
+	 * @param converter the converter to apply.
+	 * @return this.
+	 */
+	public PropertyValueConverterRegistrar registerConverter(Class<?> type, String path,
+			PropertyValueConverter<?, ?, ? extends ValueConversionContext<?>> converter) {
+
+		registry.registerConverter(type, path,
+				(PropertyValueConverter<?, ?, ? extends ValueConversionContext<P>>) converter);
+		return this;
+	}
+
+	/**
+	 * Obtain the {@link SimplePropertyValueConverterRegistry}.
+	 *
+	 * @return new instance of {@link SimplePropertyValueConverterRegistry}.
+	 */
+	public ValueConverterRegistry<P> buildRegistry() {
+		return new SimplePropertyValueConverterRegistry<>(registry);
+	}
+
+	/**
+	 * Register collected {@link PropertyValueConverter converters} within the given {@link ValueConverterRegistry
+	 * registry}.
+	 *
+	 * @return new instance of {@link SimplePropertyValueConverterRegistry}.
+	 */
+	public void registerConvertersIn(ValueConverterRegistry<P> target) {
+
+		Assert.notNull(target, "Target registry must not be null!");
+
+		registry.getConverterRegistrationMap().entrySet().forEach(entry -> {
+			target.registerConverter(entry.getKey().type, entry.getKey().path, entry.getValue());
+		});
+	}
+
+	/**
+	 * Helper to build up a fluent registration API starting on
+	 *
+	 * @author Oliver Drotbohm
+	 */
+	static class WritingConverterRegistrationBuilder<T, S, P extends PersistentProperty<P>> {
+
+		private final Consumer<PropertyValueConverter> registration;
+		private final PropertyValueConverterRegistrar config;
+
+		public WritingConverterRegistrationBuilder(Class<T> type, String property, PropertyValueConverterRegistrar config) {
+
+			this.config = config;
+			this.registration = (converter) -> config.registerConverter(type, property, converter);
+		}
+
+		<R> ReadingConverterRegistrationBuilder<T, S, S, P> writingAsIs() {
+			return writing((source, context) -> source);
+		}
+
+		<R> ReadingConverterRegistrationBuilder<T, S, R, P> writing(Function<S, R> writer) {
+			return writing((source, context) -> writer.apply(source));
+		}
+
+		/**
+		 * Describes how to convert the domain property value into the database native property.
+		 *
+		 * @param <R> the type to be written to the database
+		 * @param writer the property conversion to extract a value written to the database
+		 * @return will never be {@literal null}.
+		 */
+		<R> ReadingConverterRegistrationBuilder<T, S, R, P> writing(BiFunction<S, ValueConversionContext<P>, R> writer) {
+			return new ReadingConverterRegistrationBuilder<>(this, writer);
+		}
+	}
+
+	/**
+	 * A helper to build a fluent API to register how to read a database value into a domain object property.
+	 *
+	 * @author Oliver Drotbohm
+	 */
+	static class ReadingConverterRegistrationBuilder<T, S, R, P extends PersistentProperty<P>> {
+
+		private WritingConverterRegistrationBuilder<T, S, P> origin;
```

Comment:
Generics seem off here as `T` refers to the underlying type. Later on, we call `registration.accept` where generics do not match.